### PR TITLE
RSDK-622 - ResourceByName should check existence before returning

### PR DESCRIPTION
**Pull request by stuqdog**

### CODE REVIEW

**Comment by stuqdog** [comment]
@cheukt would love to chat with you offline briefly about this! specifically the changes to testing.

File: robot/client/client.go [hunk 1]
```diff
@@ -494,12 +495,18 @@ func (rc *RobotClient) ResourceByName(name resource.Name) (interface{}, error) {
 		return client, nil
 	}
 
-	resourceClient, err := rc.createClient(name)
-	if err != nil {
-		return nil, err
+	// before adding a new resource, make sure this name exists and is known
```
Comment (maximpertsov) [inline robot/client/client.go:498]: NIT: the comment just above (one final check) isn't quite true anymore

Comment (stuqdog) [reply robot/client/client.go]: ahh good catch! updating now :)

**Review by njooma — APPROVED** [review]

**Review by maximpertsov — APPROVED** [review]

**Review by cheukt — APPROVED** [review]
LGTM

**Pull request merged**